DataManagerPlugin.cs

```diff
@@ -183,13 +187,23 @@ string errorInfo
             var missingMods = Instance.MissingMods(saveSlot);
             if (missingMods.Count == 0)
             {
+                if (vanillaSaveIncompatibleText is { } text)
+                {
+                    SetSaveIncompatibleText(__instance, text);
+                    // Since the vanilla text is restored, we don't need to do this again.
+                    vanillaSaveIncompatibleText = null;
+                }
                 return true;
             }
             Instance.Logger.LogInfo($"save slot {saveSlot} has save data for missing mods:");
             foreach (var m in missingMods)
             {
                 Instance.Logger.LogInfo(m);
             }
+            vanillaSaveIncompatibleText = SetSaveIncompatibleText(
+                __instance,
+                new($"Mods.{DataManagerPlugin.Id}", "REQUIRED_MODS_MISSING")
```

Comment:
Minor - it's probably better to just allocate this once

Reply:
Counterpoint: this way we don't have to allocate it _at all_ unless needed, which I expect will be most of the time.